src/shared/sam/cli/samCliInvokerUtils.ts

```diff
@@ -99,3 +100,13 @@ function startsWithEscapeSequence(text: string, sequences = acceptedSequences):
 function startsWithError(text: string): boolean {
     return text.startsWith('Error:')
 }
+
+export async function addTelemetryEnvVar(options: SpawnOptions | undefined): Promise<SpawnOptions> {
+    return {
+        ...options,
+        env: {
+            SAM_CLI_TELEMETRY_FROM_IDE: await getUserAgent({ includeClientId: false }),
+            ...options?.env,
```

Comment:
related: https://github.com/aws/aws-toolkit-vscode/pull/2094

Reply:
So I guess now we probably want to conditionally disable SAM CLI telemetry based off the global Toolkit setting as SAM CLI will expect some telemetry data from the Toolkits.